plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableQuery.php

```diff
@@ -0,0 +1,657 @@
+					$this->mappings[ "{$address_type}_address" ][ $column_name ]['type']
+				);
+			}
+		}
+	}
+
+	/**
+	 * Generates the ORDER BY clause.
+	 *
+	 * @return void
+	 */
+	private function process_orderby(): void {
+		// 'order' and 'orderby' vars.
+		$this->args['order'] = $this->sanitize_order( $this->args['order'] ?? '' );
+		$this->sanitize_orderby( $this->args['orderby'] ?? 'date' );
+
+		$orderby = $this->args['orderby'];
+
+		if ( 'none' === $orderby ) {
+			$this->orderby = '';
+			return;
+		}
+
+		$orderby_array = array();
+		foreach ( $this->args['orderby'] as $_orderby => $order ) {
+			$orderby_array[] = "{$_orderby} {$order}";
+		}
+
+		$this->orderby = implode( ', ', $orderby_array );
+	}
+
+	/**
+	 * Generates the limits to be used in the LIMIT clause.
+	 *
+	 * @return void
+	 */
+	private function process_limit(): void {
+		$limit  = ( $this->arg_isset( 'limit' ) ? absint( $this->args['limit'] ) : false );
+		$page   = ( $this->arg_isset( 'page' ) ? absint( $this->args['page'] ) : 1 );
+		$offset = ( $this->arg_isset( 'offset' ) ? absint( $this->args['offset'] ) : false );
+
+		if ( ! $limit ) {
+			return;
+		}
+
+		$this->limits = array( $offset ? $offset : absint( ( $page - 1 ) * $limit ), $limit );
+	}
+
+	/**
+	 * Checks if a query var is set (i.e. not one of the "skipped values").
+	 *
+	 * @param string $arg_key Query var.
+	 * @return bool TRUE if query var is set.
+	 */
+	public function arg_isset( string $arg_key ): bool {
+		return ( isset( $this->args[ $arg_key ] ) && ! in_array( $this->args[ $arg_key ], self::SKIPPED_VALUES, true ) );
+	}
+
+	/**
+	 * Runs the SQL query.
+	 *
+	 * @return void
+	 */
+	private function run_query() {
+		global $wpdb;
+
+		// Run query.
+		$this->orders = array_map( 'absint', $wpdb->get_col( $this->sql ) ); // phpcs:ignore WordPress.DB.PreparedSQL.NotPrepared
+
+		// Set max_num_pages and found_orders if necessary.
+		if ( ( $this->arg_isset( 'no_found_rows' ) && ! $this->args['no_found_rows'] ) || empty( $this->orders ) ) {
+			return;
+		}
+
+		if ( $this->limits ) {
+			$this->found_orders  = absint( $wpdb->get_var( 'SELECT FOUND_ROWS()' ) );
+			$this->max_num_pages = (int) ceil( $this->found_orders / $this->args['limit'] );
+		} else {
+			$this->found_orders = count( $this->orders );
+		}
+	}
+
+	/**
+	 * Make some private available for backwards compatibility.
+	 *
+	 * @param string $name Property to get.
+	 * @return mixed
+	 */
+	public function __get( string $name ) {
+		switch ( $name ) {
+			case 'found_orders':
+			case 'found_posts':
+				return $this->found_orders;
+			case 'max_num_pages':
+				return $this->max_num_pages;
+			case 'posts':
+			case 'orders':
+				return $this->results;
+			default:
+				break;
+		}
+	}
+
+	/**
+	 * Returns the value of one of the query arguments.
+	 *
+	 * @param string $arg_name Query var.
+	 * @return mixed
+	 */
+	public function get( $arg_name ) {
+		return isset( $this->args[ $arg_name ] ) ? $this->args[ $arg_name ] : null;
```

Comment:
optional - you can replace this with `??` version

Reply:
Done.